roles/kubernetes-apps/external_cloud_controller/openstack/templates/external-openstack-cloud-config.j2

```diff
@@ -57,3 +57,6 @@ internal-network-name="{{ network_name }}"
 {% for network_name in external_openstack_network_public_networks %}
 public-network-name="{{ network_name }}"
 {% endfor %}
+
+[Metadata]
+search-order="{{ external_openstack_metadata_search_order }}"
```

Comment:
Can we wrap this with `if external_openstack_metadata_search_order is defined` ? and make the default value being undefined?

Reply:
Sure.

Reply:
Anyway if the value is undefined, the cloud-provider-openstack will use by default `search-order=configDrive,metadataService` as mentionned [here](https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-openstack-cloud-controller-manager.md#metadata).

Are you sure you removing the default value won't be ambiguous?

Reply:
What do you think @alijahnas ?

Reply:
The default value of configDrive,metadataService is the right one for most deployments. I am not sure in which cases these values will need to be specifically defined. Maybe for optimization.
If the default value of the CCM is the same as the one declared in kubespray-defaults, adding the line won't affect much. I am okay with @Miouge1 to add the wrapping `if external_openstack_metadata_search_order is defined` and remove the default. The commented group_vars variable is enough I think.

Reply:
Alright then, I did the changes :).